### PR TITLE
Fix function to generate message boundaries.

### PR DESCRIPTION
**Pull request by coryan**
The function to generate a message boundary was not working if the
initial string was found in the message. I found this because MSVC
issues a warning when the value of string::find() is discarded, which it
was in this case (creating an infinite loop too I think).

This is why we cleanup warnings, because sometimes they show you bugs
that you could not find otherwise.

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-cpp/1286)

### CODE REVIEW

File: google/cloud/storage/internal/generate_message_boundary_test.cc [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  auto generator = google::cloud::internal::MakeDefaultPRNG();
+
+  auto string_generator = [&generator](int n) {
+    static std::string const chars =
+        "abcdefghijklmnopqrstuvwxyz012456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    return google::cloud::internal::Sample(generator, n, chars);
+  };
+
+  // The magic constants here are uninteresting, we just want a large message
+  // and a relatives short string to start searching for a boundary.
```
Comment (houglum) [inline google/cloud/storage/internal/generate_message_boundary_test.cc]: s/relatives/relatively/

File: google/cloud/storage/internal/generate_message_boundary_test.cc [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+TEST(GenerateMessageBoundaryTest, Simple) {
+  auto generator = google::cloud::internal::MakeDefaultPRNG();
+
+  auto string_generator = [&generator](int n) {
+    static std::string const chars =
+        "abcdefghijklmnopqrstuvwxyz012456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    return google::cloud::internal::Sample(generator, n, chars);
+  };
+
+  // The magic constants here are uninteresting, we just want a large message
```
Comment (houglum) [inline google/cloud/storage/internal/generate_message_boundary_test.cc]: s/, we/. We/

File: google/cloud/storage/internal/generate_message_boundary_test.cc [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  // The initial_size and growth_size parameters are set to
+  // (kMatchedStringLength / 2) and (kMatchedStringLength / 4) respectively,
+  // that forces the algorithm to find the initial string, and to grow it
+  // several times before the kMatchedStringLength common characters are
+  // exhausted.
+  auto boundary =
+      GenerateMessageBoundary(message, std::move(string_generator),
+          kMatchedStringLength / 2, kMatchedStringLength / 4);
+  EXPECT_THAT(message, Not(HasSubstr(boundary)));
+  // We expect that
```
Comment (houglum) [inline google/cloud/storage/internal/generate_message_boundary_test.cc]: ^ This comment looks half-finished

File: google/cloud/storage/internal/generate_message_boundary_test.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/internal/generate_message_boundary_test.cc]: Fixed.

File: google/cloud/storage/internal/generate_message_boundary_test.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/internal/generate_message_boundary_test.cc]: Fixed.

File: google/cloud/storage/internal/generate_message_boundary_test.cc (hunk 3 above)
Comment (coryan) [reply google/cloud/storage/internal/generate_message_boundary_test.cc]: Fixed.

**Pull request merged**